### PR TITLE
feat: Added docker publishing

### PR DESCRIPTION
**Pull request by daviroo**
# Docker Setup and CI/CD Integration

## Changes
This PR introduces Docker support and CI/CD integration for our webhook handler project. The main changes include:

1. Added a `Dockerfile` for containerizing the application
2. Created a `.dockerignore` file to optimize Docker builds
3. Updated `README.md` with Docker-related instructions
4. Added a GitHub Actions workflow for automated Docker image building and publishing

## Dockerfile Details
- Uses Node.js 18 Alpine as the base image
- Installs all dependencies, including dev dependencies for building
- Builds the TypeScript code using `npx tsc`
- Removes dev dependencies after build for a leaner production image
- Sets up the environment for production use

## CI/CD Workflow
- Triggers on pushes to the main branch and new releases
- Builds the Docker image for both amd64 and arm64 architectures
- Pushes the image to DockerHub with appropriate tags (latest, commit SHA, and branch/tag name)
- Utilizes Docker's cache features for faster builds

## Next Steps
- Set up the required secrets (`DOCKERHUB_USERNAME` and `DOCKERHUB_TOKEN`) in the GitHub repository settings
- Test the Docker build process locally and via GitHub Actions
- Update project documentation with information about the Docker image and how to use it

## Testing
- [x] Verify local Docker build success
- [x] Confirm GitHub Actions workflow runs successfully
- [x] Test the published Docker image in a clean environment

This PR aims to streamline our deployment process and make the project more portable and easier to set up for new contributors.

### CODE REVIEW

**Comment by nully0x** [comment]
Thanks @daviroo

**Review by Extheoisah — APPROVED** [review]

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Docker Publish
+
+on:
```
Comment (nully0x) [inline .github/workflows/docker-publish.yml:3]: I would suggest we also have a dispatch trigger too

Comment (daviroo) [reply .github/workflows/docker-publish.yml:3]: @nully0x what do you envision that being used for?

Comment (nully0x) [reply .github/workflows/docker-publish.yml:3]: A dispatch helps to manually trigger action. That's useful when an action fails.

Comment (daviroo) [reply .github/workflows/docker-publish.yml:3]: AFAIK you can still rerun a failed action without a dispatch trigger.

Comment (nully0x) [reply .github/workflows/docker-publish.yml:3]: Yea I am aware we can rerun jobs.

For me dispatch helps with clean execution and also helps to distinguish events in the action workflow.

**Pull request merged**